### DISCUSSION TITLE
Current KTC compatibility and KTX 2.0 support

### DISCUSSION THREAD

**Discussion by carloshurtado-vgw** [pixijs/pixijs#7852] [Q&A]
Hi, I'm currently running PIXIjs v6 and I'm excited to try out the built-in compressed texture support.

So far I've being trying to compress a PNG file into a compatible KTX file that runs in PIXI, but haven't being successful.
I used a power of 2 texture, with a size that allows to be divided by 4 to avoid iOS possible problems.
I tried compressing the PNG image using the toktx tool from Khronos group:
https://github.com/KhronosGroup/KTX-Software/releases#:~:text=for%20issue%20%23371.-,Assets,-15
And also tried using Texture packer conversion, but no luck.

I noticed this example was posted before in another discussion:
https://pixijs.io/compressed-textures-example/

**That  example seems to successfully loads a KTX file and I would like to know which tool and settings were used to create this file so I can successfully compress my own images**.

Also going through the code of the KTX Loader I noticed that only KTX 1.0 spec is checking in the header.
**I'm wondering if PIXIjs will also include compatibility with KTX 2.0, as this seems pretty valuable**

Pixi version used: PixiJS 6.1.3

Will appreciate your help with this, thanks!

**Comment by ShukantPal** [comment]
You have to convert the PNG into a compressed format before packaging it into a KTX file. compressor.shukantpal.com let's you generate a KTX file any compressed format from a PNG.

**Reply by miltoncandelero** [reply]
What are you using under the hood to create the KTXs? 👀

**Reply by ShukantPal** [reply]
It's a basisu + libktx pipeline.

**Comment by giao123-cloud** [comment]
so how to load compressed texture in pixijs?

**Reply by bigtimebuddy** [reply]
In v6, compressed texture loading is supported by default. Simply use the Loader to load your dds, ktx, files. In order to use BASIS, you need to bootstrap the Basis WASM. Here's a very simple example:

https://github.com/pixijs/compressed-textures-example

**Comment by giao123-cloud** [comment]
thanks for your help!




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Matt ***@***.***&gt;; 
发送时间: 2022年2月22日(星期二) 凌晨0:32
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [pixijs/pixijs] Current KTC compatibility and KTX 2.0 support (Discussion #7852)





 
In v6, compressed texture loading is supported by default. Simply use the Loader to load your dds, ktx, files. In order to use BASIS, you need to bootstrap the Basis WASM. Here's a very simple example:
 
https://github.com/pixijs/compressed-textures-example
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
